compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -1260,26 +1260,27 @@ object messages {
            |""".stripMargin
   }
 
-  case class OverloadedOrRecursiveMethodNeedsResultType(method: Names.TermName)(implicit ctx: Context)
+  case class OverloadedOrRecursiveMethodNeedsResultType(methodName: Names.TermName, cycleSym: Symbol)(implicit ctx: Context)
   extends Message(OverloadedOrRecursiveMethodNeedsResultTypeID) {
     val kind = "Syntax"
```

Comment:
Oh, and we should probably classify `OverloadedOrRecursiveMethodNeedsResultType`, `RecursiveValueNeedsResultType`, `TermMemberNeedsResultTypeForImplicitSearch`, `CyclicReferenceInvolvingImplicit` and `CyclicReferenceInvolving` as `Cyclic` errors not `Syntax`